systemTests/testImageUpload.py

```diff
@@ -1,7 +1,43 @@
 import unittest
-from controller.controller import controller
+import subprocess
+import time
+from shellScripts.scriptWrapper import ScriptWrapper
 
 
-class TestImageUpload(unittest.testcase):
-    def run_controller_and_handler_and_upload_image(self):
-        pass
+class TestImageUpload(unittest.TestCase):
+    _UPLOAD_STATUS_SUCCESS = "{\"upload status\": \"file upload sucessful\"}"
+
+    def test_run_controller_and_handler_and_upload_image(self):
+        """
+        Calls scripts from shellScripts as done in normal execution.
+        """
+        # run the controller
+        print("~~~ Running the controller ~~~")
+        controller = subprocess.Popen(ScriptWrapper.runControllerPath())
+        # wait for controller to instantiate
+        time.sleep(2)
+
+        # run Flask Server
+        print("~~~ Running the Flask Server ~~~")
+        flask_server = subprocess.Popen(ScriptWrapper.runFlaskServerPath())
+        # wait for Flask Server and controller to connect
+        time.sleep(5)
+
+        # initiate image upload via curl command
+        print("~~~ Calling curl command ~~~")
+        try:
+            output = subprocess.check_output(ScriptWrapper.curlCommandPath(), timeout=25)
+        except TimeoutExpired:
+            self.fail("Image upload timed out!")
```

Comment:
I think there might be a memory leak here.
Do we need to call `controller.kill()` or `flask_server.kill()`, when these Exceptions or assertions are called or terminate the program and test? 

On an aside: For non-integration tests, `controller.kill()` and `flask_server.kill()` can go in a tear down method that the potentially offered by the Python unit test framework.